lib/import/colorectal/providers/st_george/st_george_handler_colorectal.rb

```diff
@@ -0,0 +1,317 @@
+
+            genes_dict = process_genes(record)
+            handle_test_status(record, genotype, genes_dict)
+          end
+
+          def process_genes(record)
+            # process the gene names in columns listed and check for matches with the CRC gene regex list
+            # add matched tgenes to the genes_dict
+            # if no matched to crc gene regex list then check for matches in CRC gene map to add to the genes_dict
+            # return genes_dict
+            genes_dict = {}
+
+            ['gene', 'gene (other)', 'variant dna', 'test/panel'].each do |column|
+              genes = []
+              gene_list = record.raw_fields[column]&.scan(CRC_GENE_REGEX)
+
+              gene_list = process_test_panels(record, gene_list, column) if column == 'test/panel'
+
+              next if gene_list.nil?
+
+              gene_list.each do |gene|
+                CRC_GENE_MAP[gene]&.each do |gene_value|
+                  genes.append(gene_value)
+                end
+              end
+
+              genes_dict[column] = genes.uniq
+            end
+            genes_dict
+          end
+
+          def process_test_panels(record, gene_list, column)
+            # extracts panels tested from record
+            # map panels to list of genes in FULL_SCREEN_TESTS_MAP
+            # return list of genes tested in panel
+            panel_genes_list = FULL_SCREEN_TESTS_MAP[record.raw_fields['test/panel']]
+            panel_genes_list&.each do |gene|
+              gene_list.append(gene)
+            end
+
+            r211 = record.raw_fields[column]&.eql?('R211')
+
+            if r211.present?
+              r211_genes = process_r211(record)
+              r211_genes.each do |gene|
+                gene_list.append(gene)
+              end
+            end
+
+            gene_list
+          end
+
+          def process_r211(record)
+            # identify genes tested on the R222 panel based on authorised date
+            # return list of genes in R211 panel
+            return unless record.raw_fields['test/panel'] == 'R211'
+
+            date = DateTime.parse(record.raw_fields['authoriseddate'])
+
+            if date < DateTime.parse('18/07/2022')
+              r211_panel_genes = %w[APC BMPR1A EPCAM MLH1 MSH2 MSH6 MUTYH NTHL1 PMS2 POLD1 POLE PTEN
+                                    SMAD4 STK11]
+            elsif date >= DateTime.parse('18/07/2022')
+              r211_panel_genes = %w[APC BMPR1A EPCAM MLH1 MSH2 MSH6 MUTYH NTHL1 PMS2 POLD1 POLE PTEN
+                                    SMAD4 STK11 GREM1 RNF43]
+            end
+            r211_panel_genes
+          end
+
+          def handle_test_status(record, genotype, genes)
+            # Creates a duplicate genotype for each gene
+            # Determines whether test is targeted or full screen and runs duplicate_genotype method.
+            # Return genotypes list
+            if genotype.targeted?
```

Comment:
I think this could be simplified a little, something like:

```
            columns = if genotype.targeted?
                        ['gene', 'gene (other)', 'variant_dna', 'variant protein', 'test/panel']
                      elsif genotype.full_screen?
                        ['gene', 'gene (other)', 'variant_dna', 'test/panel']
                      end

            duplicate_genotype(columns, genotype, genes, record) if columns.present?
```

Reply:
Simplified logic in [075175e](https://github.com/NHSDigital/data_management_system/commit/075175e335e5a81d699eebf4501ad28f9d5a8048)